dijkstra-prim-visualization/src/components/Canvas/Canvas.jsx

```diff
@@ -1,115 +1,115 @@
-import { useState, useRef } from "react";
+import { useState, useRef, useContext } from "react";
 import classes from "./Canvas.module.css";
-import Node from "./Node/Node";
-import Edge from "./Edge/Edge";
 import { newEdgeValid, newNodePositionValid } from "./CanvasUtils";
+import { GraphParamsContext } from "../../GraphParamsContext";
+import Nodes from "./Nodes/Nodes";
+import Edges from "./Edges/Edges";
+
+const MAX_EDGE_WEIGHT = 100;
 
 /**
  * Canvas component for visualizing nodes and edges.
+ * Handles user interactions for adding nodes and edges
  * @returns {JSX.Element} The Canvas component.
  */
 const Canvas = () => {
-  // State variables to keep track of all nodes and edges
-  const [nodePoints, setNodePoints] = useState([]);
-  const [edges, setEdges] = useState([]);
+  // Destructure the states from context
+  const { nodes, setNodes, edges, setEdges } = useContext(GraphParamsContext);
 
-  // Object with default data to reset firstNode, when needed
-  const resetNodeData = {
+  // Object with default data to reset firstClickedNode, when needed
+  const resetFirstClickedNode = {
     isClicked: false,
-    x: null,
-    y: null,
+    node: null,
   };
 
-  // State variable to keep track of the first node clicked when creating an edge
-  const [firstNode, setFirstNode] = useState(resetNodeData);
+  // Initialize state to track the first clicked node
+  const [firstClickedNode, setFirstClickedNode] = useState(
+    resetFirstClickedNode,
+  );
 
   // Reference to the canvas SVG element
   const canvasRef = useRef(null);
 
-  // Handler function for when the canvas is clicked
+  /**
+   * Handler function for when the canvas is clicked.
+   * Adds a new node at the clicked position if the position is valid.
+   * @param {MouseEvent} event - The click event.
+   */
   const canvasClickHandler = (event) => {
-    // Reset the firstNode state variable
-    setFirstNode(resetNodeData);
+    setFirstClickedNode(resetFirstClickedNode);
 
     // Calculate the coordinates of the clicked point relative to the canvas
-    const nodePointAbsoluteX = event.clientX;
-    const nodePointAbsoluteY = event.clientY;
-    const nodePointCanvasRelativeX =
-      nodePointAbsoluteX - canvasRef.current.getBoundingClientRect().left;
-    const nodePointCanvasRelativeY =
-      nodePointAbsoluteY - canvasRef.current.getBoundingClientRect().top;
-
-    // Create a new node point object
-    const newNodePoint = {
-      x: nodePointCanvasRelativeX,
-      y: nodePointCanvasRelativeY,
+    const nodeAbsoluteX = event.clientX;
+    const nodeAbsoluteY = event.clientY;
+    const nodeCanvasRelativeX =
+      nodeAbsoluteX - canvasRef.current.getBoundingClientRect().left;
+    const nodeCanvasRelativeY =
+      nodeAbsoluteY - canvasRef.current.getBoundingClientRect().top;
+
+    const newNode = {
+      id: `${nodeCanvasRelativeX}:${nodeCanvasRelativeY}`,
```

Comment:
Would it be possible to use zero-based indices as node IDs? E.g. first node has id = 0, second id = 1, etc. In algorithms, we use integer IDs so it would be nice to avoid mapping back and forth if component IDs can be ints as well.

Reply:
sure, by tomorrow evening will be done